### ISSUE TITLE
Suspended not playable in Frotz

### ISSUE THREAD

**Issue by eriktorbjorn** [AlexProudfoot/The-ZIL-Files#68]
Suspended is currently not playable. The status line is messed up, and input doesn't work. It seems that whatever you type it gets added to what's already in the input buffer.

Note, though, that the game was never officially released as a Z5 game so this may represent a work in progress.

**Comment by eriktorbjorn** [comment]
I spent some time trying to get the new status line to work, and found several problems:

- It uses DIROUT incorrectly. It needs to explicitly turn off output to screen, not just set it to output to a buffer.
- It prints too many spaces when updating the first row, overwriting the "Casualties this cycle" text.
- It uses ```<TELL D .L>``` to print the robot's location, which produces garbage. I think that should be ```<TELL SD .L>``` instead.

All this is, of course, easily fixable. But as it stands, I don't see any way to fit the status line within the traditional 80 columns. It prints three robots's locations on each line. I think the longest room names are 24 characters ("Hydroponics Control Area" and "Vehicle Debarkation Area"), so fitting three of those in one line uses 72 characters already. That only leaves 8 characters for the robot names.

And of course it gets even worse if we ever want to support narrower screens.

Considering how broken the game is, perhaps it would be more sensible to roll back to use an older version of the source code?

**Comment by eriktorbjorn** [comment]
I made a quick attempt at compiling the older version of the source code, using some of the fixes you made to the later one. It seems to work a lot better.

By the way, the PEOPLE object defines the CLC-TXT property twice. I believe that you've disabled the wrong one in the current version, because in the original game you can only look up the humans in the Advisory Pedestal. It will say "These humans can be of help in reaching areas which cannot be reached by other means."

**Comment by eriktorbjorn** [comment]
Of course, there may still be changes in the newer version worth incorporating in the older, if we go that path:

- In the FRANKLIN object, the word "futher" has been corrected to "further".
- In the I-CAR-WAIT routine, "his plug" and "his clone" were changed to "the plug" and "the clone".
- In I-STARCROSS "We've given him more than enough time." has been changed to "There's been more than enough time."
- In PARRIVE8 "then give him some time" has been changed to "then see if things return to normal".

I'm guessing that the changes were made to show that they consider you a piece of faulty equipment, rather than a human.

Anyway, that's just the textual changes I've found. I can't easily determine if the rest of the changes are relevant or not.

**Comment by AlexProudfoot** [comment]
What newer/older versions are you talking about? I can only find one version of the sources in the original repository. They have been converted, at least partially, to XZIP. The status line is more elaborate than the original ZIP version. The status line and the input line are messed up. ZARF has published two versions of the sources. Neither of them compile as published but the second (called rlater) has  some changes which look helpful.

**Comment by AlexProudfoot** [comment]
Actually, the first (called r8) has the original status line and seems to work better. I'm going to substitute it for the original.

**Comment by AlexProudfoot** [comment]
This looks much the same as the original r8-s830521 except (strangely) the score starts at 3.

![Untitled](https://user-images.githubusercontent.com/49520257/59550912-4d232e00-8f69-11e9-8d9d-f24e317b7840.png)

**Comment by eriktorbjorn** [comment]
Yes, Zarf's "r8" source code was what I meant when I said "older version". I didn't compare his "rlater" source code to the one Jason Scott leaked, but I assumed they were the same.

I was able to get the "r8" code to build by using some of the changes made to get the version here to build. I didn't save mine, because it was just a quick-and-dirty experiment, so I don't remember what "score" I started out with.

As I mentioned, there's a typo in "r8" ("futher" instead of "further"), in addition to the two typos that have already been fixed here. As for the other textual changes between "r8" and "rlater", I'll leave that for you to decide.

**Comment by eriktorbjorn** [comment]
This bug has been fixed in The Infocom Files, by using Release 8 instead.
